### PR TITLE
Refactor Tests by Functionality

### PR DESCRIPTION
**Pull request by lgaleana-llm**
Refactor the tests by grouping them based on the functionality they are testing. Ensure that no file exceeds 100 lines of code.

### Test Plan

pytest / playwright

### CODE REVIEW

**Comment by lgaleana-llm** [comment]
Total time to create PR: 193.35 seconds
Total OPenAI cost: 0.9350950000000001

File: test_api.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import pytest
```
Comment (lgaleana) [inline test_api.py]: Put all test related stuff in a `tests` folder

Comment (lgaleana-llm) [reply test_api.py]: Sure, I'll move all the test-related files into a `tests` folder. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
Total time amending the PR: 127.86 secondsTotal OPenAI cost: 0.45520499999999997

**Review by lgaleana — APPROVED** [review]

**Pull request merged**